packages/server/dataloader/azureDevOpsLoaders.ts

```diff
@@ -0,0 +1,425 @@
+              return null
+            }
+            const provider = await parent.get('integrationProviders').loadNonNull(providerId)
+
+            const manager = new AzureDevOpsServerManager(
+              azureDevOpsAuthToRefresh,
+              provider as IntegrationProviderAzureDevOps
+            )
+            const oauthRes = await manager.refresh(refreshToken)
+            if (oauthRes instanceof Error) {
+              return null
+            }
+            const {accessToken, refreshToken: newRefreshToken} = oauthRes
+            const updatedRefreshToken =  newRefreshToken || refreshToken
+            const newAzureDevOpsAuth = {
+              ...azureDevOpsAuthToRefresh,
+              accessToken,
+              refreshToken: updatedRefreshToken
+            }
+            await upsertTeamMemberIntegrationAuth(newAzureDevOpsAuth)
+            return newAzureDevOpsAuth
+          }
+          return azureDevOpsAuthToRefresh
+        })
+      )
+      return results.map((result) => (result.status === 'fulfilled' ? result.value : null))
+    },
+    {
+      ...parent.dataLoaderOptions,
+      cacheKeyFn: (key) => `${key.userId}:${key.teamId}`
+    }
+  )
+}
+
+export const azureDevOpsAllWorkItems = (
+  parent: RootDataLoader
+): DataLoader<TeamUserKey, WorkItem[] | undefined, string> => {
+  return new DataLoader<TeamUserKey, WorkItem[] | undefined, string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
+          const returnWorkItems = [] as WorkItem[]
+          const auth = await parent
+            .get('teamMemberIntegrationAuths')
+            .load({service: 'azureDevOps', teamId, userId})
+
+          if (!auth) {
+            return undefined
+          }
+
+          const provider = await parent.get('integrationProviders').loadNonNull(auth.providerId)
+
+          const manager = new AzureDevOpsServerManager(
+            auth,
+            provider as IntegrationProviderAzureDevOps
+          )
+
+          const restResult = await manager.getAllUserWorkItems(null, false)
+          const {error, workItems} = restResult
+          if (error !== undefined || workItems === undefined) {
+            console.log(error)
+          } else {
+            returnWorkItems.push(...workItems)
+          }
+          return returnWorkItems
+        })
+      )
+      return results.map((result) => (result.status === 'fulfilled' ? result.value : undefined))
+    },
+    {
+      ...parent.dataLoaderOptions,
+      cacheKeyFn: (key) => `${key.teamId}:${key.userId}`
+    }
+  )
+}
+
+export const azureDevUserInfo = (
+  parent: RootDataLoader
+): DataLoader<TeamUserKey, AzureUserInfo | undefined, string> => {
+  return new DataLoader<TeamUserKey, AzureUserInfo | undefined, string>(
+    async (keys) => {
+      const results = await Promise.allSettled(
+        keys.map(async ({userId, teamId}) => {
+          const auth = await parent
```

Comment:
-1 Why not use `const auth = await parent.get('freshAzureDevOpsAuth').load({teamId, userId})`?

Reply:
changed